lib/lib.es5.d.ts

```diff
@@ -1258,6 +1258,12 @@ interface Array<T> {
       * @param callbackfn A function that accepts up to three arguments. The filter method calls the callbackfn function one time for each element in the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
+    filter<S extends T>(callbackfn: (value: T, index: number, array: T[]) => value is S, thisArg?: any): S[];
```

Comment:
As for ReadonyArray, this method is already there. For typed arrays, it doesn't make sense to me (since there is no type inheritance like for generic arrays).